File: contracts/wasm/gascalibration/memory/go/memoryimpl/funcs.go
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

package memoryimpl

import "github.com/iotaledger/wasp/packages/wasmvm/wasmlib/go/wasmlib"

func funcF(ctx wasmlib.ScFuncContext, f *FContext) {
	n := f.Params.N().Value()
	store := make([]uint32, n)
	for i := uint32(0); i < n; i++ {
		store[i] = i
	}
}

File: contracts/wasm/gascalibration/memory/ts/memoryimpl/funcs.ts
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import * as wasmlib from "wasmlib";
import * as wasmtypes from "wasmlib/wasmtypes";
import * as sc from "../memory/index";

export function funcF(ctx: wasmlib.ScFuncContext, f: sc.FContext): void {
    const n = f.params.n().value();
    const store: u32[] = [];
    for (var i: u32 = 0; i < n; i++) {
        store.push(i);
    }
}
